src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -75,7 +75,9 @@ def filter(self, dataset, times, intermediate_step=None, linearise_at=None):
             rvs.append(filtrv)
         return KalmanPosterior(times, rvs, self, with_smoothing=False)
 
-    def filter_step(self, start, stop, current_rv, data, intermediate_step=None):
+    def filter_step(
+        self, start, stop, current_rv, data, intermediate_step=None, diffusion=1.0
```

Comment:
so the filter() method gets a diffusion constant? Why exactly? Isnt this something we want to set to 1 all the time and calibrate secretly, instead of asking the user to specify it at filter time?

Reply:
Good thing you comment on this! To be honest, I was not really sure in which parts of Kalman you want to have the diffusion, so I added it to `predict`, `smooth_step` (and related) and `filter_step`. I guess we do need it for the smoothing, right? Therefore it felt natural to also add it into `filter_step`, but I don't really know.

Reply:
I think that this is wise, but I suppose on filter(), smooth() and filtsmooth() we do not want it (cp. with ODESolver.solve(), where we probably dont want it either). Instead, in those functions, I'd argue that it is the calibration's job to handle sigma. What do you think?

Reply:
What do you mean with "it is the calibration's job to handle the sigma"? I don't really understand where we want to get / how we want to pass the diffusions down, if not by function arguments.

Reply:
Everything I meant was that I dont think it is a good idea to allow a user to change  the sigma behaviour while calling filter() or filtsmooth(), because we want to nudge a user towards "trusting" the calibration. Is that more clear? Sorry, I am aware this might not be the most clear explanation. if you have more questions, we can talk in person?

Reply:
Just to note what we talked about on Friday: `filter_step`, `predict` and `smooth_step` should indeed have a diffusion parameter, whereas `filter`, `smooth`, `smooth_list` and `filtsmooth` should not, so I would leave things as they were. Though if I misunderstood something or if you disagree please re-open this and I'll change things :)

Reply:
Fine